functions/common/src/oracle.ts

```diff
@@ -8,47 +8,37 @@ import type {AuthProof} from "./types";
 
 const genRequestId = async function(
     provider: Provider,
-    nameHash: string,
-    func: string,
-    data: string | [],
+    owner: string,
+    func: string
 ) {
   const hexlink = await hexlContract(provider);
   const result = ethers.utils.keccak256(
-      ethers.utils.defaultAbiCoder.encode(
-          ["bytes4", "bytes", "address", "uint256", "uint256"],
-          [
-            func,
-            data,
-            hexlink.address,
-            (await provider.getNetwork()).chainId,
-            await hexlink.nonce(nameHash),
-          ]
-      )
+    ethers.utils.defaultAbiCoder.encode(
+      ["bytes4", "address", "uint256", "address"],
+      [
+        func,
+        hexlink.address,
+        (await provider.getNetwork()).chainId,
+        owner
```

Comment:
in our latest contract impl, the last param is "bytes memory" which is the result of `encodeFunctionData("init", owner);` https://github.com/HexlinkOfficial/hexlink-contracts-evm/blob/main/test/account.ts#L8